src/codegen/compile.ml

```diff
@@ -5600,17 +5667,23 @@ module MakeSerialization (Strm : Stream) = struct
       ReadBuf.get_ptr get_data_buf ^^ set_old_pos ^^
 
       let go' can_recover env t =
-        ( (* Reset depth counter if we made progress *)
-          ReadBuf.get_ptr get_data_buf ^^ get_old_pos ^^
-          G.i (Compare (Wasm.Values.I32 I32Op.Eq)) ^^
-          G.if1 I32Type
-          (get_depth ^^ compile_add_const 1l)
-          (compile_unboxed_const 0l)
-        ) ^^
-        (if can_recover
-         then compile_unboxed_const 1l
-         else get_can_recover) ^^
-        deserialize_go env t
+        (* assumes idltyp on stack *)
+        Stack.with_frame env "frame_ptr" 3l (fun () ->
+          Stack.set_local env 0l ^^ (* arg get_idltyp *)
+            (* set up frame arguments *)
+            ( (* Reset depth counter if we made progress *)
+              ReadBuf.get_ptr get_data_buf ^^ get_old_pos ^^
+              G.i (Compare (Wasm.Values.I32 I32Op.Eq)) ^^
+              G.if1 I32Type
+                (Stack.get_prev_local env 1l ^^ compile_add_const 1l)
```

Comment:
Detail: Maybe the frame offsets of the specific local variables could be defined as constants.

Reply:
Good idea.

Reply:
Done